### PR TITLE
chore: prepend aws-amplify to cli-core

### PR DESCRIPTION
**Pull request by evcodes**
#### Description of changes
starting the work for amplify-cli-core -> aws-amplify/amplify-cli-core. Migrated auth and geo packages to test the changes.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli-core-vNext/lib/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export * from 'amplify-cli-core';
+//# sourceMappingURL=index.d.ts.map
```
Comment (sobolk) [inline packages/amplify-cli-core-vNext/lib/index.d.ts:2]: please remove `lib` folder and add it to gitignore file

File: packages/amplify-cli-core-vNext/package.json [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    "@types/lodash": "^4.14.149",
+    "@types/node": "^12.12.6",
+    "@types/rimraf": "^3.0.0",
+    "@types/uuid": "^8.0.0",
+    "@types/yarnpkg__lockfile": "^1.1.5",
+    "nock": "^13.0.11",
+    "rimraf": "^3.0.0",
+    "strip-ansi": "^6.0.0",
+    "uuid": "^8.3.2"
+  },
```
Comment (sobolk) [inline packages/amplify-cli-core-vNext/package.json:74]: this should depend only on cli-core and maybe on rimraf.

**Review by sobolk — CHANGES_REQUESTED** [review]

**Review by sobolk — APPROVED** [review]
LGTM. Please get clean (minus known failures) e2e run for this before merging.

**Review by awsluja — COMMENTED** [review]
Looks good but need a passing e2e run before merge

File: packages/amplify-cli-core/package.json [hunk 3]
```diff
@@ -7,6 +7,9 @@
     "url": "https://github.com/aws-amplify/amplify-cli.git",
     "directory": "packages/amplify-cli-core"
   },
+  "publishConfig": {
```
Comment (akshbhu) [inline packages/amplify-cli-core/package.json:10]: why is this added ?. I think we only need this for scoped packages

**Review by akshbhu — APPROVED** [review]
Looks good , Do we need a ```.npmIgnore``` file in ```amplify-cli-core-vNext```?

File: packages/amplify-cli-core/package.json (hunk 3 above)
Comment (sobolk) [reply packages/amplify-cli-core/package.json:10]: Staging ground for future work. see https://github.com/aws-amplify/amplify-cli/pull/12235#discussion_r1137502614 .

**Comment by sobolk** [comment]
> Looks good but need a passing e2e run before merge

Fwiw, he's got pretty good result here https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/17421/workflows/d34a2f7f-3431-4341-adb7-175db7bedc62 . Just missing merge from dev where we fixed that custom migration test.

**Comment by sobolk** [comment]
> Looks good , Do we need a `.npmIgnore` file in `amplify-cli-core-vNext`?

We don't . that folder will go away in next release cycle where we prefix actual cli core and remove this shell package.

**Review by awsluja — APPROVED** [review]
